src/otx/algo/classification/torchvision_model.py

```diff
@@ -0,0 +1,250 @@
+            layers = list(last_layer.children())[:-1]
+            self.head = nn.Sequential(*layers, nn.Linear(feature_channel, num_classes))
+        else:
+            feature_channel = last_layer.in_features
+            self.head = nn.Linear(feature_channel, num_classes)
+
+        self.softmax = nn.Softmax(dim=-1)
+        self.loss = nn.CrossEntropyLoss() if loss is None else loss
+
+    def forward(
+        self,
+        images: torch.Tensor,
+        labels: torch.Tensor | None = None,
+        mode: str = "tensor",
+    ) -> torch.Tensor:
+        """Performs forward pass of the model.
+
+        Args:
+            images (torch.Tensor): The input images.
+            labels (torch.Tensor): The ground truth labels.
+            mode (str, optional): The mode of the forward pass. Defaults to "tensor".
+
+        Returns:
+            torch.Tensor: The output logits or loss, depending on the training mode.
+        """
+        feats = self.backbone(images)
+        if len(feats.shape) == 4:  # If feats is a 4D tensor: (batch_size, channels, height, width)
+            feats = feats.view(feats.size(0), -1)  # Flatten the output of the backbone: (batch_size, features)
+        logits = self.head(feats)
+        if mode == "tensor":
+            return logits
+        if mode == "loss":
+            return self.loss(logits, labels)
+        return self.softmax(logits)
+
+
+class OTXTVModel(OTXMulticlassClsModel):
+    """OTXTVModel is that represents a TorchVision model for multiclass classification.
+
+    Args:
+        backbone (
+            Literal["resnet50", "efficientnet_b0", "efficientnet_b1", "efficientnet_b3", "efficientnet_b4",
+            "efficientnet_v2_l", "mobilenet_v3_small"]):
+            The backbone architecture of the model.
+        num_classes (int): The number of classes for classification.
+        loss (nn.Module | None, optional): The loss function to be used. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        backbone: Literal[
+            "resnet50",
+            "efficientnet_b0",
+            "efficientnet_b1",
+            "efficientnet_b3",
+            "efficientnet_b4",
+            "efficientnet_v2_l",
+            "mobilenet_v3_small",
+        ],
+        num_classes: int,
+        loss: nn.Module | None = None,
+    ) -> None:
+        self.backbone = backbone
+        self.loss = loss
+
+        super().__init__(num_classes=num_classes)
+
+    def _create_model(self) -> nn.Module:
+        return TVModelWithLossComputation(
+            backbone=self.backbone,
+            num_classes=self.num_classes,
+            loss=self.loss,
+        )
+
+    def _customize_inputs(self, inputs: MulticlassClsBatchDataEntity) -> dict[str, Any]:
+        if isinstance(inputs.images, list):
+            images = tv_tensors.wrap(torch.stack(inputs.images, dim=0).to(dtype=torch.float32), like=inputs.images[0])
+        else:
+            images = inputs.images.to(dtype=torch.float32)
```

Comment:
Do not manually change input tensor's `dtype` to have `dtype=torch.float32` here. Lightning Trainer is responsible for it according to `Trainer(precision=...)`.

Reply:
Thanks - Done https://github.com/openvinotoolkit/training_extensions/pull/3034/commits/1e693cf8a3b5ed5519946989f35ad3b9115966ea